### ISSUE TITLE
Can't use it inside Gatsby

### ISSUE THREAD

**Issue by animanoir** [martinlaxenaire/curtainsjs#74]
It seems it renders some HTML stuff, but it actually doesn´t shows anything at all.

**Comment by thibault-mahe** [comment]
Hi @animanoir, I managed without much trouble to make it work with a gatsby website using [react-curtains](https://github.com/martinlaxenaire/react-curtains). Would you mind sharing some code showing how you implemented your work ?

best,
Thibault

**Comment by animanoir** [comment]
I jave an `index.js` file where I put the `SimplePlane` component. I wrap it inside the `Curtains` component. But it doesn´t renders at all.

```
import SimplePlane from '../components/SimplePlane/SimplePlane.js'
import { Curtains } from "react-curtains";

          <Curtains pixelRatio={Math.min(1.5, window.devicePixelRatio)}>
            <SimplePlane />
          </Curtains>
```

**Comment by martinlaxenaire** [comment]
Hi @animanoir ,

just in case, what CSS styles are you applying to both your curtains and plane components?
Also, are there any warnings/errors logged to the console?

If you could set up a live example this would be ideal to help debug this issue.

Cheers,

**Comment by ceiphr** [comment]
Hello,

Same issue. I've written this component:

```js
import React from 'react';
import { Curtains, Plane } from 'react-curtains';

const basicVs = `
    precision mediump float;
    
    attribute vec3 aVertexPosition;
    attribute vec2 aTextureCoord;
    
    uniform mat4 uMVMatrix;
    uniform mat4 uPMatrix;
    
    uniform mat4 uTextureMatrix0;
    
    varying vec3 vVertexPosition;
    varying vec2 vTextureCoord;
    
    void main() {
        gl_Position = uPMatrix * uMVMatrix * vec4(aVertexPosition, 1.0);
        
        // varyings
        vVertexPosition = aVertexPosition;
        vTextureCoord = (uTextureMatrix0 * vec4(aTextureCoord, 0.0, 1.0)).xy;
    }
`;


const basicFs = `
    precision mediump float;

    varying vec3 vVertexPosition;
    varying vec2 vTextureCoord;
    
    uniform sampler2D uSampler0;
    
    uniform float uTime;
    
    void main() {
        vec2 textureCoord = vTextureCoord;
        // displace our pixels along the X axis based on our time uniform
        // textures coords are ranging from 0.0 to 1.0 on both axis
        textureCoord.x += sin(textureCoord.y * 25.0) * cos(textureCoord.x * 25.0) * (cos(uTime / 50.0)) / 25.0;
        
        gl_FragColor = texture2D(uSampler0, textureCoord);
    }
`;

const BasicPlane = ({children}) => { 
    const basicUniforms = {
        time: {
            name: "uTime",
            type: "1f",
            value: 0
        }
    };

    const onRender = (plane) => {
        plane.uniforms.time.value++;
    };

    return (
        <Curtains>
            <Plane
                className="BasicPlane"

                // plane init parameters
                vertexShader={basicVs}
                fragmentShader={basicFs}
                uniforms={basicUniforms}

                // plane events
                onRender={onRender}
            >
                {children}
            </Plane>
        </Curtains>
    )
}

export default BasicPlane
```

It is based on https://github.com/martinlaxenaire/react-curtains#basic-example. No CSS styles applied to the plane or canvas. No errors are produced by using the component. The child image (tested with webP and PNG) is rendered and that's it. Nothing happens. 

I've only installed https://github.com/martinlaxenaire/react-curtains. I'm testing on the latest versions of Chrome and Firefox.

On a related note, [the example page is completely blank for me](https://www.curtainsjs.com/examples/basic-plane/index.html).

Please let me know if anymore details are neccessary.

**Comment by martinlaxenaire** [comment]
Hi @ceiphr,

first of all, thanks for the curtainsjs basic plane bug report, there was indeed an issue with it which should be fixed in v8.0.1 - 8181a052c45a1f77091d260d851d0dd33ea88d19


As for the react-curtains issue, you should apply CSS styles to both your canvas container (it usually covers the whole viewport) and your plane HTML element (to define its size and hide its image element). See this codesandbox https://codesandbox.io/s/react-curtains-basic-plane-h30ie?file=/src/styles.css

I've updated the react-curtains documentation to make it clearer. In case this does not solve your issue, I encourage you to create an issue in the react-curtains repo.

Cheers,

**Comment by martinlaxenaire** [comment]
Closing this issue as it seems to be solved.
